pkg/util/kubernetes/kubernetes.go

```diff
@@ -128,6 +128,12 @@ func CreateMgmtService(kubecli corev1client.CoreV1Interface, clusterName, cluste
 			TargetPort: intstr.FromInt(constants.MonitoringPort),
 			Protocol:   v1.ProtocolTCP,
 		},
+		{
+			Name:       "metrics",
+			Port:       constants.MetricsPort,
```

Comment:
It'll add a metrics port regardless whether metrics are enabled or not on the crd.

While this isn't optimal, the current code doesn't give the option to update the service after it is created. Meaning if someone enables metrics after deploying a nats cluster, it won't be updated.

---
Additionally AFAIK k8s doesn't add the pods to the service/metrics endpoint which do not expose their metrics pod.

Therefore if metrics are disabled, no endpoints will be added, which is the correct behaviour.